### PR TITLE
Fixed exception when pid list is outdated

### PR DESCRIPTION
**Pull request by albertogomcas**
I added the fix for https://github.com/MrYsLab/python_banyan/issues/32

Additionally I made some fixes so all the tests pass.

### CODE REVIEW

File: python_banyan/banyan_base_multi/banyan_base_multi.py [hunk 1]
```diff
@@ -32,6 +32,7 @@
 import msgpack_numpy as m
 import zmq
 import os
+import umsgpack
```
Comment (MrYsLab) [inline python_banyan/banyan_base_multi/banyan_base_multi.py:35]: I am not sure why you are importing umsgpack. The [msgpack package is being used ](https://github.com/MrYsLab/python_banyan/blob/4ff4b95c58cb0476d3363407f2ea161f8d061457/python_banyan/banyan_base_multi/banyan_base_multi.py#L31) currently in this file as all others. 
Is there a problem using msgpack for this file?

Comment (albertogomcas) [reply python_banyan/banyan_base_multi/banyan_base_multi.py:35]: umsgpack is used when self.numpy is false

https://github.com/MrYsLab/python_banyan/blob/4ff4b95c58cb0476d3363407f2ea161f8d061457/python_banyan/banyan_base_multi/banyan_base_multi.py#:~:text=default%3Dm.encode)-,else%3A,message%20%3D%20umsgpack.packb(payload),-pub_envelope%20%3D%20topic.encode

If if its not imported then these two tests fail (at least on my machine?) 
https://github.com/MrYsLab/python_banyan/blob/4ff4b95c58cb0476d3363407f2ea161f8d061457/python_banyan/tests/banyan_base_multi/test_banyanBaseMulti.py#:~:text=def%20test_publish_payload_invalid_socket(self,assert%20True

**Comment by albertogomcas** [comment]
After comparing with the similar conditional in BanyanBase:

https://github.com/MrYsLab/python_banyan/blob/4ff4b95c58cb0476d3363407f2ea161f8d061457/python_banyan/banyan_base/banyan_base.py#L207-L211

 I see no reason to keep umsgpack in BanyanBaseMulti, so I removed the extra import and just used msgpack everywhere

**Comment by MrYsLab** [comment]
@albertogomcas I have merged the changes and uploaded version 3.11 to pypi. Thanks for the PR.

**Pull request merged**